### PR TITLE
Add StdIn as an acceptable argument for apply, preview, destroy, and diff commands

### PR DESCRIPTION
**Pull request by seans3**
* Adds StdIn as an acceptable argument to the following commands: `apply`, `preview`, `destroy`, and `diff`.
* Updates `Use:` in command
* Adds unit tests
* Tested manually

### CODE REVIEW

File: pkg/common/path.go [hunk 1]
```diff
@@ -14,30 +14,43 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
 )
 
+const stdinDash = "-"
```
Comment (mortent) [inline pkg/common/path.go:17]: I'm not sure we should use this convention. I know we do it with kubectl, but I think we can just assume that no arguments means stdin.

Comment (seans3) [reply pkg/common/path.go:17]: Right now, both "no arguments" and "dash" mean StdIn. If you think it  should only be "no arguments", please let me know.

Comment (mortent) [reply pkg/common/path.go:17]: I think I slightly prefer that we don't support the dash argument. It is easier to add it later than to remove it once it is in.

Comment (seans3) [reply pkg/common/path.go:16]: Removed dash as an argument; only "no argument" means stdin now.

File: pkg/common/path.go [hunk 2]
```diff
@@ -13,32 +13,43 @@ import (
...
 	}
 
+	// Must be a single directory here; set recursive flag.
 	t := true
 	fileNameFlags.Filenames = &paths
 	fileNameFlags.Recursive = &t
 	return fileNameFlags
 }
 
+// DemandOneDirectoryOrStdin processes "paths" to ensure the
```
Comment (seh) [inline pkg/common/path.go:34]: This doesn't match the function name `DemandOneDirectory`. Did you intend to rename the function to match the comment?

File: cmd/diff/cmddiff.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ import (
 func NewCmdDiff(f util.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
 	options := diff.NewDiffOptions(ioStreams)
 	cmd := &cobra.Command{
-		Use:                   "diff DIRECTORY",
+		Use:                   "diff (DIRECTORY | STDIN)",
```
Comment (seh) [inline cmd/diff/cmddiff.go:22]: I find that this fails in what appears to be `common.ExpandPackageDir` and `common.expandDir`:

> % ... | kpt live diff   
> error: lstat -: no such file or directory

It looks like we need to thread this treatment of the single hyphen (meaning standard input) deeper into the call path.

Comment (seh) [reply cmd/diff/cmddiff.go:22]: See #170.

**Pull request merged**